kolibri/core/logger/api.py

```diff
@@ -1053,3 +1057,68 @@ class AttemptLogViewSet(ReadOnlyValuesViewset):
     filter_class = AttemptFilter
 
     values = attemptlog_values
+
+
+class GenerateCSVLogRequestSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = GenerateCSVLogRequest
+        fields = [
+            "facility",
+            "log_type",
+            "selected_start_date",
+            "selected_end_date",
+            "date_requested",
+        ]
+
+
+class GenerateCSVLogRequestFilter(FilterSet):
+    log_type = ChoiceFilter(choices=GenerateCSVLogRequest.LOG_TYPE_CHOICES)
+    facility = ModelChoiceFilter(queryset=Facility.objects.all())
+
+    class Meta:
+        model = GenerateCSVLogRequest
+        fields = ["log_type", "facility"]
+
+
+class GenerateCSVLogRequestViewSet(viewsets.ModelViewSet):
+    """
+    API endpoint that allows csv log request data to be created or updated
+    """
+
+    permission_classes = (KolibriAuthPermissions,)
```

Comment:
Please follow up after this is merged with any updates here to ensure only facility admins and super users can create these new models. I think using `KolibriAuthPermissionsFilter` similar to other viewsets here, along with some `permissions = RoleBasedPermissions` defined on the new model should do it.